client/state/sites/plans/test/selectors.js

```diff
@@ -53,6 +55,74 @@ describe( 'selectors', () => {
 			expect( plans ).to.eql( plans1 );
 		} );
 	} );
+	describe( '#getCurrentPlan()', () => {
+		context( 'when no plan data is found for the given siteId', () => {
+			const siteId = 99999999;
+			const state = {
+				sites: {
+					plans: {
+						[ siteId ]: {}
+					}
+				}
+			};
+
+			it( 'returns null', () => {
+				const plan = getCurrentPlan( state, siteId );
+				expect( plan ).to.eql( null );
+			} );
+		} );
+
+		context( 'when plans are found for the given siteId', () => {
+			const siteId = 99999999;
+			const plan1 = { currentPlan: true };
+			const state = {
+				sites: {
+					plans: {
+						[ siteId ]: {
+							data: [ plan1 ]
+						}
+					},
+					items: {
+						[ siteId ]: {
+							URL: 'https://example.wordpress.com'
+						}
+					}
+				}
+			};
+
+			context( 'when those plans include a \'currentPlan\'', () => {
+				before( () => {
+					plan1.currentPlan = true;
+				} );
+
+				it( 'returns the currentPlan', () => {
+					const plan = getCurrentPlan( state, siteId );
+					expect( plan ).to.eql( plan1 );
+				} );
+			} );
+
+			context( 'when those plans do not include a \'currentPlan\'', () => {
+				let selectors, newSitePlanObject;
+
+				useMockery( ( mockery ) => {
+					mockery.registerMock( './assembler', {
+						createSitePlanObject: () => newSitePlanObject
```

Comment:
Oh hah taking a second look, and I don't think we want that fallback.

@retrofox I think we'll want to remove the fallback in getCurrentPlan.
```javascript
	// get Site plan from the `site` data
	const site = sitesList.getSelectedSite();
	const plan = createSitePlanObject( site.plan );
	debug( 'current plan: %o', plan );
	return plan;
```
For one it's using sitesList 😢  and it's we're passing through a site.plan object through to a /sites/plans assembler. I think some unexpected things will occur. Any callers can instead use the appropriate selectors in the other state trees.